### PR TITLE
Include 0 as max in elastic axis if all values are negative

### PR DESCRIPTION
**Pull request by sebgrohn**
For Row chart and charts using Stack mixin, use the corresponding method for determining the max value for the y axis as used for min value:

> `yAxisMin`, current behaviour: If all values are _positive,_ use 0 as y axis _min._
> `yAxisMax`, new behaviour: If all values are _negative,_ use 0 as y axis _max._

Should fix issue #879.

Two Jasmine tests are currently failing, both related to Composite chart and its `alignYAxes`. As far as I can tell, it seems to be related to the calculations in [`calculateYAxisRanges`](https://github.com/dc-js/dc.js/blob/develop/src/composite-chart.js#L109) that does something magic with the y axis ranges if there are two y axes and `alignYAxes` are set. I am at loss here what to do, all suggestions are welcome!

[jasmine-failing-tests.txt](https://github.com/dc-js/dc.js/files/302730/jasmine-failing-tests.txt)

### CODE REVIEW

**Comment by gordonwoodhull** [comment]
@sebgrohn, this looks like an excellent approach. I like the symmetry of it.

Would you be willing to also write/improve a few tests for this? 

In particular, the row chart [already has tests for all-negative values](https://github.com/dc-js/dc.js/blob/develop/spec/row-chart-spec.js#L359),  but it looks like it doesn't test `elasticX`. 

The bar chart [apparently does test elasticY with negative values](https://github.com/dc-js/dc.js/blob/develop/spec/bar-chart-spec.js#L507), but it didn't detect any problems. [Ditto for the line chart.](https://github.com/dc-js/dc.js/blob/develop/spec/line-chart-spec.js#L510)

Don't worry about `alignYAxes` since that feature is still somewhat in flux and #1033 may help there.

**Comment by sebgrohn** [comment]
Great! I will try to produce some tests for negative values with elasticY enabled. I can't give any time frame right now, though.

**Comment by sebgrohn** [comment]
Added tests for Row, Bar, and Line chart that checks for the correct x/y axis ticks when `elasticX` / `elasticY` is enabled.

*Notes:*
* I added the missing Line chart tests for enabled `elasticX`, all three cases (all positive, mixed, all negative). I'm not sure, however, if they should be included in the template function `itShouldBehaveLikeARowChartWithGroup` or not...
* The Bar and Line chart tests only tested with mixed-sign values, so I renamed them and added new ones for all-negative values.

File: spec/row-chart-spec.js [hunk 1]
```diff
@@ -352,11 +352,30 @@ describe('dc.rowChart', function () {
...
+                        chart.elasticX(true)
+                            .xAxis().ticks(3);
+
+                        chart.render();
+                    });
+
+                    it('should generate x axis domain dynamically', function () {
+                        var nthText = function (n) { return d3.select(chart.selectAll('g.axis .tick text')[0][n]); };
+
+                        for (let i = 0; i < xAxisTicks.length; i++) {
```
Comment (gordonwoodhull) [inline spec/row-chart-spec.js]: Crazy... all your code works well, but for some reason the `let` here caused Jasmine to skip the row chart tests! Completely silent failure!

dc.js is still ES5 and will be for a while yet. @sebgrohn, did you have to do anything special to get the tests to run?

Of course, I noticed 58 commits later that my test count was way down. Easy enough to fix, but whoah that's a little scary if one ES6 keyword causes a file to get dropped entirely.

**Comment by gordonwoodhull** [comment]
Thanks @sebgrohn, rebased and merged in 2.0 beta 32!

File: spec/row-chart-spec.js (hunk 1 above)
Comment (sebgrohn) [inline spec/row-chart-spec.js]: Wow, that's interesting (and scary)! Sloppy to let a `let` (sic.) slip in there; I am truly ES6 damaged.

I don't remember if I did something out of the ordinary, and I have the code and dev env set up on another computer I can't access right now... :o

**Comment by sebgrohn** [comment]
Great!
